src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php

```diff
@@ -19,18 +19,27 @@
  *
  * @author Christophe Coevoet <stof@notk.org>
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Diego Saint Esteben <diego@saintesteben.me>
  */
 class DecoratorServicePass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
+        $definitions = new \SplPriorityQueue();
```

Comment:
The usage of a priority query is a BC break here. Currently, decorators with the same priority (i.e. all of them for now) are applied in the order in which they appear in ``$container->getDefinitions()``, which is more or less the order in which services have been defined (it is a bit more complex when a service gets overridden actually). but a SplPriorityQueue does not retain the order of insertion for the same priority.

And this is inconsistent with all other places using priorities in Symfony as we preserve the insertion order when the priority is the same.